modules/internal/ChapelBase.chpl

```diff
@@ -1909,7 +1909,12 @@ module ChapelBase {
 
   /* swap operator */
   inline proc <=>(ref lhs, ref rhs) {
-    const tmp = lhs;
+    // it's tempting to make this a `const` tmp, but it causes
+    // problems for cases that self-modify, like records wrapping
```

Comment:
It's not modifying itself - it's modifying `tmp` on the line `rhs = tmp`.

Reply:
@mppf: How do you feel about the phrasing "but it causes problems for types that self-modify when they appear on the RHS of an assignment, like records wrapping..."?

Reply:
how about

> but it causes problems for types where the RHS of an assignment is modified, as with records containing an `owned` field

The part I was not liking is the phrase "self-modify".